common/src/chain/transaction/account_outpoint.rs

```diff
@@ -41,6 +47,12 @@ impl From<AccountSpending> for AccountType {
 pub enum AccountSpending {
     #[codec(index = 0)]
     Delegation(DelegationId, Amount),
+    #[codec(index = 1)]
+    TokenUnrealizedSupply(TokenId, Amount),
```

Comment:
Please explain above each arm what it does... not sure I understand.

Reply:
It lacks a FIXME to search for a better name. The idea is that I need to spend from an account in both cases: when supply is increased as well as decreased. So I introduced 2 spending types that represent spending from supply which is not circulating (called it unrealized) which increases supply and from circulating supply which decreases it. 

Same idea applies to locking the supply. I added another spending type to be able to authorize locking.

Reply:
After looking at it again I find it confusing indeed and even diverging from our current design approaches.

Here `TxInput` type is used to determine the type of an operation. For example to decrease supply `AccountSpending::TokenCirculatingSupply` -> `TxOutput::Burn` transaction must be created. 

To make it more aligned with current codebase I could use `TxOutput`. In this case `Account::Token` -> `TxOutput::DecreaseSupply` + `TxOutput::Burn` tx would be used. 

I'll rework it by adding new `TxOutput` types instead. Also I think I'll add one nesting level to the `TxOutput` enum to simplify code. So it would look like `TxOutput::Token::IssueToken` or `TxOutput::Token::IncreaseSupply`